tests/ecmp/test_inner_hashing.py

```diff
@@ -0,0 +1,190 @@
+    duthost.fetch(src="/tmp/fib.{}.txt".format(timestamp), dest="/tmp/fib")
+
+    po = config_facts.get('PORTCHANNEL', {})
+    ports = config_facts.get('PORT', {})
+
+    tmp_fib_info = tempfile.NamedTemporaryFile()
+    with open("/tmp/fib/{}/tmp/fib.{}.txt".format(duthost.hostname, timestamp)) as fp:
+        fib = json.load(fp)
+        for k, v in fib.items():
+            skip = False
+            prefix = k.split(':', 1)[1]
+            ifnames = v['value']['ifname'].split(',')
+            nh = v['value']['nexthop']
+
+            oports = []
+            for ifname in ifnames:
+                if po.has_key(ifname):
+                    oports.append([str(mg_facts['minigraph_ptf_indices'][x]) for x in po[ifname]['members']])
+                else:
+                    if ports.has_key(ifname):
+                        oports.append([str(mg_facts['minigraph_ptf_indices'][ifname])])
+                    else:
+                        logger.info("Route point to non front panel port {}:{}".format(k, v))
+                        skip = True
+            # skip direct attached subnet
+            if nh == '0.0.0.0' or nh == '::' or nh == "":
+                skip = True
+
+            if not skip:
+                tmp_fib_info.write("{}".format(prefix))
+                for op in oports:
+                    tmp_fib_info.write(" [{}]".format(" ".join(op)))
+                tmp_fib_info.write("\n")
+            else:
+                tmp_fib_info.write("{} []\n".format(prefix))
+    tmp_fib_info.flush()
+
+    ptfhost.copy(src=tmp_fib_info.name, dest=FIB_INFO_FILE_DST)
+    msg = "Copied FIB info to PTF host '{}': local_path={}, remote_path={}"
+    logger.info(msg.format(ptfhost.hostname, tmp_fib_info.name, FIB_INFO_FILE_DST))
+
+    tmp_fib_info.close()
+
+
+@pytest.fixture(scope='module')
+def vlan_ptf_ports(config_facts, tbinfo):
+    ports = []
+    for vlan_members in config_facts.get('VLAN_MEMBER', {}).values():
+        for intf in vlan_members.keys():
+            dut_port_index = config_facts.get('port_index_map', {})[intf]
+            logging.info("Added " + str(dut_port_index))
+            ports.append(dut_port_index)
+
+    return ports
+
+
+@pytest.fixture(scope='module')
+def router_mac(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    return duthost.facts['router_mac']
+
+
+@pytest.fixture(scope="module")
+def hash_keys():
+    hash_keys = HASH_KEYS[:]
+    return hash_keys
+
+
+@pytest.fixture(scope="module")
+def symmetric_hashing(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    symmetric_hashing = False
+
+    if duthost.facts['asic_type'] in ["mellanox"]:
+        symmetric_hashing = True
+
+    return symmetric_hashing
+
+
+@pytest.fixture(params=["ipv4", "ipv6"])
+def ipver(request):
+    return request.param
+
+
+def test_inner_hashing(hash_keys, ptfhost, ipver, router_mac, vlan_ptf_ports, symmetric_hashing, build_fib, setup):
```

Comment:
If this test is intended for Mellanox platform only at this stage. please add proper platform check and skip test when platform is not within the short list.

Reply:
Makes sense, added pytest.mark.asic('mellanox') at the beginning of the test